File: 2562. Find the Array Concatenation Value/Solution.py
class Solution:
    def findTheArrayConcVal(self, nums: List[int]) -> int:
        res, left, right = 0, 0, len(nums) - 1
        while left < right:
            res += nums[left] * (10 ** self.findDigits(nums[right])) + nums[right]
            left += 1
            right -= 1
        if left == right: res += nums[left]
        return res

    def findDigits(self, x):
        d = 0
        while x > 0:
            x //= 10
            d += 1
        return d

File: 2562. Find the Array Concatenation Value/Solution.java
class Solution {
    public long findTheArrayConcVal(int[] nums) {
        long res = 0;
        int n = nums.length, left = 0, right = n - 1;
        while (left < right) {
            int shift = digits(nums[right]);
            res += nums[left] * Math.pow(10, shift) + nums[right];
            left++;
            right--;
        }
        if (left == right) res += nums[left];
        return res;
    }
    public int digits(int x) {
        int d = 0;
        while (x > 0) {
            x /= 10;
            d++;
        }
        return d;
    }
}